docs/contracts/NFT Contract/NFT Actions/purchase.a.md

```diff
@@ -0,0 +1,140 @@
+
+The transferred uniqs will automatically be moved to the `transfer_tokens_receiver_account` that was setup during the purchase requirements setup.
+
+Internally we are constructing a vector of which `token_ids` to be transferred.
+
+
+## Action Parameters
+
+**Action Interface**
+
+| Property Name    | C++ Type                            | JavaScript Type | Description                                                                                                                                                                         |
+| ---------------- | ----------------------------------- | --------------- | ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| token_factory_id | uint64_t                            | number          | ID of a token factory to purchase from                                                                                                                                              |
+| index            | uint64_t                            | number          | Index of purchase option to use                                                                                                                                                     |
+| max_price        | asset                               | string          | Maximum amount of UOS you allow to be withdrawn from your account. If price is set in USD this will prevent transaction from overcharging you in case USD price goes down           |
+| buyer            | eosio::name                         | string          | Account that will pay UOS and/or Uniqs for this purchase                                                                                                                            |
+| receiver         | eosio::asset                        | string          | Account that will receive the Uniq from this purchase                                                                                                                               |
+| promoter_id      | std::optional\<eosio::name>         | string / null   | Optional promoter of the purchase transaction. If no promoter is provided then the default promoter specified in `saleshrlmcfg` (scope `0`) will be used if present                 |
+| user_uniqs       | std::optional\<provided_user_uniqs> | Array / null    | List of uniqs the buyer is willing to provide for this purchase option to either be taken from him or to just verify their presence. Refer to `provided_user_uniqs` breakdown below |
+| memo             | std::string                         | string          | A short operation description                                                                                                                                                       |
+
+**provided_user_uniqs Interface**
```

Comment:
provided_user_uniqs Interface is a vector of uniqs_count_internal, which is the following sturcuture